src/control/cmd/daos_agent/security_rpc_test.go

```diff
@@ -61,13 +61,12 @@ func TestAgentSecurityModule_HandleCall_BadMethod(t *testing.T) {
 	defer common.ShowBufferOnFailure(t, buf)
 
 	mod := NewSecurityModule(log, nil)
-	resp, err := mod.HandleCall(newTestSession(t, log, &net.UnixConn{}), -1, nil)
-
-	if resp != nil {
-		t.Errorf("Expected no response, got %+v", resp)
+	method, err := mod.ID().GetMethod(-1)
```

Comment:
you can't pass an invalid method ID into HandleCall now, that's one of the benefits of this restructuring, because the methods are now typed you can't supply an invalid one. This test has been changed to now verify an error is returned when you attempt to lookup and return a method when supplying an invalid ID.

Reply:
HandleCall still performs the check, though. It seems like we'd still want to test that behavior in unit tests even if it can't be hit in the greater context. The unit tests are in context of the specific module, after all. IMO the new types should have their own unit tests closer to where they live.